lib/plsql.go

```diff
@@ -676,6 +676,8 @@ func (fun Function) prepareCall() (decls, pre []string, call string, post []stri
 							k, *arg.TableOf)
 
 						if arg.IsInput() {
+							pre = append(pre,
```

Comment:
Can't do this unconditionally - it's required for NESTED TABLE, but forbidden for ASSOCIATIVE ARRAY:

```
DECLARE
  TYPE pls_tt IS TABLE OF PLS_INTEGER INDEX BY PLS_INTEGER;
  v_tt pls_tt := pls_tt();  --not needed, but ok
  TYPE pls_nt IS TABLE OF PLS_INTEGER;
  v_nt pls_nt := pls_nt();  --required
BEGIN
  v_nt.EXTEND;  --required
  v_nt(1) := 1;
  DBMS_OUTPUT.PUT_LINE( v_nt.COUNT );
  --v_tt.EXTEND;  --forbidden
  v_tt(1) := 1;
  DBMS_OUTPUT.PUT_LINE( v_tt.COUNT );
END;
```

so we must distinguish them.

Reply:
For help, "SELECT DISTINCT coll_type FROM all_plsql_coll_types" shows "PL/SQL INDEX TABLE", "VARRAY" and "TABLE". The latter is the nested table (which needs instantiation and EXTEND), the first is the associative array which does not need instantiation and forbids EXTEND.

Another distinct feature for them is then INDEX_BY column in all_plsql_coll_types: it's empty for nested tables and not empty for associative arrays. This information is collected in [main.go](https://github.com/tgulacsi/oracall/blob/master/main.go#L857) but not used.

I've added IndexBy field to Argument, see b9ab77c .

Reply:
Thanks :) , just committed this not sure it will work, have nt tested it for both cases. just extending when the type is "TABLE". kept the initialization for both cases